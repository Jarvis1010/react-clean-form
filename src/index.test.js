import React, { Fragment } from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";
import sinon from "sinon";
import Form from "./index";

describe("Form functions", () => {
  describe("form renders with truthy validator", () => {
    let wrapper;
    let spy;
    beforeEach(() => {
      spy = sinon.spy();
      wrapper = shallow(
        <Form
          initialState={{ input: "" }}
          submitValidator={() => true}
          onSubmit={spy}
        >
          {state => (
            <Fragment>
              <input name="input" value={state.input} />
              <button type="submit">Submit</button>
            </Fragment>
          )}
        </Form>
      );
    });
    it("Renders component", () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it("calls custom onSubmit", () => {
      wrapper
        .find("form")
        .first()
        .simulate("submit");

      expect(spy.calledOnce).toBe(true);
    });
    it("updates value of input", () => {
      const newValue = "Hello";
      wrapper
        .find("form")
        .first()
        .simulate("change", { target: { name: "input", value: newValue } });
      expect(wrapper.state().input).toBe(newValue);
    });
  });

  describe("form render with falsey validator", () => {
    let wrapper;
    let spy;
    beforeEach(() => {
      spy = sinon.spy();
      wrapper = shallow(
        <Form submitValidator={() => false} onSubmit={spy}>
          {() => (
            <Fragment>
              <input />
              <button type="submit">Submit</button>
            </Fragment>
          )}
        </Form>
      );
    });
    it("Renders component", () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it("does not call onSubmit", () => {
      wrapper
        .find("form")
        .first()
        .simulate("submit");

      expect(spy.calledOnce).toBe(false);
      expect(wrapper.state().submitError).toBe(true);
    });
  });
});
