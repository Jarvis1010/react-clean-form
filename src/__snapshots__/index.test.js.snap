// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form functions Renders component 1`] = `
<form
  onChange={[Function]}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    key=".0"
    name="input"
    value=""
  />
  <select
    key=".1"
    name="select"
    value=""
  >
    <option
      key=".0"
      value="value1"
    >
      Value1
    </option>
    <option
      key=".1"
      value="value2"
    >
      Value2
    </option>
  </select>
  <select
    key=".2"
    multiple={true}
    name="selectMultiple"
    value={Array []}
  >
    <option
      key=".0"
      value="value1"
    >
      Value1
    </option>
    <option
      key=".1"
      value="value2"
    >
      Value2
    </option>
  </select>
  <input
    checked={false}
    key=".3"
    name="radio"
    type="radio"
    value="radio1"
  />
   Radio1
  <input
    checked={false}
    key=".5"
    name="radio"
    type="radio"
    value="radio2"
  />
   Radio2
  <input
    checked={false}
    key=".7"
    name="checkbox1"
    type="checkbox"
    value="checkbox1"
  />
   checkbox1
  <input
    checked={false}
    key=".9"
    name="checkbox2"
    type="checkbox"
    value="checkbox2"
  />
   checkbox2
  <button
    key=".b"
    type="submit"
    value=""
  >
    Submit
  </button>
</form>
`;
